packages/flutter/lib/src/cupertino/tab.dart

```diff
@@ -0,0 +1,161 @@
+///
+/// See also:
+///
+///  * [CupertinoTabScaffold] a typical host that supports switching between tabs.
+///  * [CupertinoPageRoute] a typical modal page route pushed onto the [CupertinoTab]'s
+///    [Navigator].
+class CupertinoTab extends StatelessWidget {
+  const CupertinoTab({
+    Key key,
+    this.home,
+    this.routes,
+    this.onGenerateRoute,
+    this.onUnknownRoute,
+    this.navigatorObservers,
+  }) : super(key: key);
+
+  /// The widget for the default route of the tab ([Navigator.defaultRouteName],
+  /// which is `/`).
+  ///
+  /// If [home] is specified, then [routes] must not include an entry for `/`,
+  /// as [home] takes its place.
+  final Widget home;
+
+  /// This tab's routing table.
+  ///
+  /// When a named route is pushed with [Navigator.pushNamed] inside this tab,
+  /// the route name is looked up in this map. If the name is present,
+  /// the associated [WidgetBuilder] is used to construct a [CupertinoPageRoute]
+  /// that performs an appropriate transition to the new route.
+  ///
+  /// If the tab only has one page, then you can specify it using [home] instead.
+  ///
+  /// When pushing named routes from [home] via [Navigator.of], be sure to use
+  /// the [BuildContext] of this tab rather than this tab's ancestors by using
+  /// a [Builder] widget.
+  ///
+  /// If [home] is specified, then it implies an entry in this table for the
+  /// [Navigator.defaultRouteName] route (`/`), and it is an error to
+  /// redundantly provide such a route in the [routes] table.
+  ///
+  /// If a route is requested that is not specified in this table (or by
+  /// [home]), then the [onGenerateRoute] callback is called to build the page
+  /// instead.
+  ///
+  /// This routing table is not shared with any routing tables of ancestor or
+  /// descendent [Navigator]s.
+  final Map<String, WidgetBuilder> routes;
+
+  /// The route generator callback used when the tab is navigated to a named route.
+  ///
+  /// This is used if [routes] does not contain the requested route.
+  final RouteFactory onGenerateRoute;
+
+  /// Called when [onGenerateRoute] also fails to generate a route.
+  ///
+  /// This callback is typically used for error handling. For example, this
+  /// callback might always generate a "not found" page that describes the route
+  /// that wasn't found.
+  ///
+  /// The default implementation pushes a route that displays an ugly error
+  /// message.
+  final RouteFactory onUnknownRoute;
+
+  /// The list of observers for the [Navigator] created in this tab.
+  ///
+  /// This list of observers is not shared with ancestor or descendent [Navigator]s.
+  final List<NavigatorObserver> navigatorObservers;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Navigator(
+      onGenerateRoute: _onGenerateRoute,
+      onUnknownRoute: _onUnknownRoute,
+      observers: navigatorObservers ?? const <NavigatorObserver>[], // Can't be null.
```

Comment:
this default should be in the constructor, not here

Reply:
done